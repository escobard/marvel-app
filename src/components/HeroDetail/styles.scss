@import "../../styles/variables";

.hero-detail {
	h2 {
		text-align: center;
		padding: $space-xlrg;
	}
	.stats {
		width: 96.4%;
		display: inline-block;
		border-top: solid 1px #003c8f;
		margin: 1.8%;
		padding: $space-xlrg 0;
		border-bottom: solid 1px #003c8f;
		.stat {
			text-align: center;
		}
	}
	.bottom {
		display: inline-block;
		width: 100%;
		padding-bottom: $space-xlrg;
		.breadcrumbs {
			float: left;
		}
		.quick-links {
			padding: 0 !important;
			float: right;
		}
	}
}

@media screen and (max-width: $break-medium) {
	.hero-detail {
		h2{
			padding: $space-med;
		}
		.stats {
			.stat {
				border-bottom: solid 1px #003c8f;
				margin-bottom: $space-lrg;
				padding-bottom: $space-lrg;
				&:nth-child(4){
					border-bottom: 0;
					margin-bottom: 0;
					padding-bottom: 0;
				}
			}
		}
		.bottom {
			padding-bottom: $space-med;
			.breadcrumbs {
				float: none;
				margin: auto;
				width: fit-content;
			}
			.quick-links {
				padding: 0 !important;
				float: right;
			}
		}
	}
}