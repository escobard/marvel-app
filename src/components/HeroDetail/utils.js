import React from "react";

export const stat = stat => {
	if (stat.available) {
		return stat.available;
	}
	return <p>N / A</p>;
};