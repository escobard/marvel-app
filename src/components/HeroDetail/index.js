import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import {
	Card,
	CardActions,
	CardHeader,
	CardMedia,
	CardTitle,
	CardText
} from "material-ui/Card";
import FlatButton from "material-ui/FlatButton";

import styles from "./styles.scss";

import { img } from "../../utils";
import { stat } from "./utils";

export default class HeroDetail extends Component {
	renderHero = hero => {
		let {
			comics,
			events,
			description,
			name,
			series,
			stories,
			urls,
			thumbnail: { path, extension },
			loading
		} = this.props.hero.name ? this.props.hero : { thumbnail: {} };
		return (
			<div>
				<Card>
					<CardMedia>
						<img
							src={img(path, "landscape_incredible", extension)}
							alt={`${name}'s image`}
						/>
					</CardMedia>
					<h1>{name}</h1>
					<CardText>
						{description ? description : "No description available"}
					</CardText>
					<div className="stats">
						<div className="stat col-md-3">
							<h4>Events</h4>
							{stat(events)}
						</div>
						<div className="stat col-md-3">
							<h4>Comics</h4>
							{stat(comics)}
						</div>
						<div className="stat col-md-3">
							<h4>Series</h4>
							{stat(series)}
						</div>
						<div className="stat col-md-3">
							<h4>Stories</h4>
							{stat(stories)}
						</div>
					</div>
					<div className="bottom">
						<div className="breadcrumbs">
							<Link to="/" className="link">
								<FlatButton label="Home" />
							</Link>
							<span> / </span>
							<Link to={`/hero/${name}`} className="link">
								<FlatButton label={name} />
							</Link>
						</div>
						<CardActions className="quick-links">
							{urls.map((url, index) => {
								return (
									<a key={index} href={url.url}>
										<FlatButton label={url.type} />
									</a>
								);
							})}
						</CardActions>
					</div>
				</Card>
			</div>
		);
	};

	render() {
		return (
			<article className="hero-detail">
				{this.props.hero ? this.renderHero(this.props.hero) : null}
			</article>
		);
	}
}

HeroDetail.defaultProps = {
	hero: {}
};

HeroDetail.propTypes = {
	hero: PropTypes.object
};