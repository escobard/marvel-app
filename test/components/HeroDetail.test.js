import React, { Component } from 'react'
import HeroDetail from '../../src/components/HeroDetail';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('HeroDetail snapshot', () => {
	const initialState = {output:100}
	const mockStore = configureStore()
	let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<HeroDetail /> )  
    })

    it('+++ render the HeroDetail component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});