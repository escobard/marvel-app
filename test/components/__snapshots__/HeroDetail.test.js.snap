// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroDetail snapshot +++ render the HeroDetail component 1`] = `
<div
  className="hero-list"
>
  <p>
    test
  </p>
</div>
`;
